phonenumber_field/widgets.py

```diff
@@ -4,42 +4,52 @@
 from django.forms.widgets import MultiWidget
 from django.utils import translation
 from phonenumbers import PhoneNumberFormat
-from phonenumbers.data import _COUNTRY_CODE_TO_REGION_CODE
 from phonenumbers.phonenumberutil import (
-    national_significant_number,
+    COUNTRY_CODE_TO_REGION_CODE,
     region_code_for_number,
 )
 
-from phonenumber_field.phonenumber import PhoneNumber
+from phonenumber_field.phonenumber import PhoneNumber, to_python
 
 try:
     import babel
 except ModuleNotFoundError:
     babel = None
 
+# ISO 3166-1 alpha-2 to national prefix
+REGION_CODE_TO_COUNTRY_CODE = {
+    region_code: country_code
+    for country_code, region_codes in COUNTRY_CODE_TO_REGION_CODE.items()
+    for region_code in region_codes
+}
+
+
+def localized_choices(language):
+    if babel is None:
+        raise ImproperlyConfigured(
+            "The PhonePrefixSelect widget requires the babel package be installed."
+        )
```

Comment:
This exception in my opinion belongs to choices preparation. The `babel` library purpose is purely to translate country names. I moved choices related logic from `__init__()` method because it was a bit complex and because of testing which will follow this a bit latter if you let me contribute more. At some stage I would love to change `PhonePrefixSelect.__init__()` declaration to match it's parent class, to accept choices argument as requested in #458. If developers using our library wants to pass own choices it's up to them to provide strings translated or not. Therefore this check would not required here but it would be mandatory there.

Reply:
Makes total sense, thanks for explaining! :+1: